pkg/network/setup/netconf.go

```diff
@@ -121,7 +122,9 @@ func (c *NetConf) Setup(vmi *v1.VirtualMachineInstance, networks []v1.Network, l
 		netpod.WithCacheCreator(c.cacheCreator),
 		netpod.WithBindingPlugins(c.clusterConfigurer.GetNetworkBindings()),
 		netpod.WithLogger(log.Log.Object(vmi)),
-		netpod.WithVMIIfaceStatuses(vmi.Status.Interfaces),
+		netpod.WhenDynamicPodInterfaceNamingEnabled(c.clusterConfigurer,
+			netpod.WithVMIIfaceStatuses(vmi.Status.Interfaces),
+		),
```

Comment:
I prefer we do not inject this info down to the `netpod`, it should not care about this.

Can this be done?
```go
vmiIfacesStatuses := vmi.Status.Interfaces
if !cluterConfig.DynamicPodInterfaceNamingEnabled() {
    vmiIfacesStatuses = nil
}

netpod := netpod.NewNetPod(
    ...
    netpod.WithVMIIfaceStatuses(vmiIfacesStatuses),
)
```

Reply:
That was my idea from the beginning, but though It was not going to be well received.